### PR TITLE
[18.0][MIG] base_technical_user: Migration to 18.0

### PR DESCRIPTION
**Pull request by Kimkhoi3010**
Note:

- Missing commits are ported: [[17.0][FW] base_technical_user: port from 14.0](https://github.com/OCA/server-tools/pull/3068)

### CODE REVIEW

File: base_technical_user/models/models.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+                self_sudoer = self_sudoer.with_context(
+                    allowed_company_ids=self.env.company.ids
+                )
+        elif raise_if_missing:
+            raise UserError(
+                _("The technical user is missing in the company {}").format(
+                    self.env.company.name
+                )
+            )
+        return self_sudoer
```
Comment (henrybackman) [inline base_technical_user/models/models.py]: Only version 14 has this modified version of this, 15, 16 and 17 are like this:
```
    def sudo_tech(self, raise_if_missing=False):
        self_sudoer = self
        tech_user = self.env.user.company_id.user_tech_id
        if tech_user:
            self_sudoer = self.with_user(tech_user.id)
        elif raise_if_missing:
            raise UserError(
                _("The technical user is missing in the company {}").format(
                    self.env.user.company_id.name
                )
            )
        return self_sudoer
```

File: base_technical_user/readme/CREADITS.md
Comment (henrybackman) [inline base_technical_user/readme/CREADITS.md:1]: Misspelled file name, should be `CREDITS.md`

Comment (Kimkhoi3010) [reply base_technical_user/readme/CREADITS.md:1]: That was my big mistake. Thank you for the reminder.

File: base_technical_user/models/models.py (hunk 1 above)
Comment (Kimkhoi3010) [reply base_technical_user/models/models.py]: Thank you for the reminder.

**Comment by simahawk** [comment]
@Kimkhoi3010 pls check the CI and fixups

**Review by simahawk — APPROVED** [review]

**Comment by Kimkhoi3010** [comment]
@simahawk, @henrybackman 
I created a new PR [OCA#3074](https://github.com/OCA/server-tools/pull/3074) to OCA/server-tools.
(This PR was closed auto after rebased)

**Pull request closed without merging**